// Copyright 2019 Google LLC. All Rights Reserved.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
syntax = "proto3";

package tfx.components.pusher;

// Defines the destination of pusher.
message PushDestination {
  oneof destination {
    Filesystem filesystem = 3;
  }

  reserved 1, 2, 4, 5;

  // File system based destination definition.
  message Filesystem {
    // The directory that Pusher will copy model exports to. The pipeline owner
    // must have write permission on this directory.
    string base_directory = 1;

    reserved 2,3;
  }
}
